flink-core/src/test/java/org/apache/flink/testutils/assertj/CauseTreeContainsAssert.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.testutils.assertj;
+
+import org.apache.flink.util.ExceptionUtils;
+
+import org.assertj.core.api.AbstractThrowableAssert;
+import org.assertj.core.api.InstanceOfAssertFactory;
+
+import java.util.function.Predicate;
+
+/**
+ * {@code CauseTreeContainsAssert} extends {@link AbstractThrowableAssert} by Flink-specific
+ * assertions.
+ */
+public class CauseTreeContainsAssert
```

Comment:
Should we come up with a different name? Something more "generic" to the exceptions. If we decide to add new methods in the future, that do different thing than "contains", it could be misleading

Reply:
👍 I switched to `FlinkThrowableAssert`